File: atinject-api/src/main/java/org/atinject/api/authentication/dto/LoginRequest.java
package org.atinject.api.authentication.dto;

import javax.validation.constraints.NotNull;

import org.atinject.core.nullanalysis.NonNull;
import org.atinject.core.websocket.dto.WebSocketRequest;

public class LoginRequest extends WebSocketRequest {

    private static final long serialVersionUID = 1L;

    @NotNull private String username;
    @NotNull private String password;
    
    @NonNull
    public String getUsername() {
        return username;
    }
    
    public LoginRequest setUsername(@NonNull String username) {
        this.username = username;
        return this;
    }
    
    @NonNull
    public String getPassword() {
        return password;
    }
    
    public LoginRequest setPassword(@NonNull String password) {
        this.password = password;
        return this;
    }
    
    
}

File: atinject-integration/src/main/resources/webapp/scripts/atinject/authentication/LoginRequest.js
define(["atinject/WebSocketRequest"],
function(WebSocketRequest){
    function LoginRequest() {
        this._class = "org.atinject.api.authentication.dto.LoginRequest"
    };

    LoginRequest.prototype = new WebSocketRequest();

        LoginRequest.prototype.getUsername = function() {
        return this.username;
    };

    LoginRequest.prototype.setUsername = function(username) {
        this.username = username;
        return this;
    };

    LoginRequest.prototype.getPassword = function() {
        return this.password;
    };

    LoginRequest.prototype.setPassword = function(password) {
        this.password = password;
        return this;
    };


    
    return LoginRequest;
});
